presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -271,4 +284,39 @@ public Expression rewriteLogicalBinaryExpression(LogicalBinaryExpression node, N
                     .collect(toImmutableList());
         }
     }
+
+    private static class GroupingOperationContext
```

Comment:
do you need a wrapping class for simple boolean? It would be fine to just have a plain boolean variable with a good name: `isGroupIdNodePresent`